File: leetcode/1easy/이경호/661.cpp
class Solution {
public:
  vector<vector<int>> imageSmoother(vector<vector<int>>& img) {
    vector<vector<int>> newImg(img.size(), vector<int>(img[0].size(), 0));
    for (int i = 0; i < img.size(); i++) {
      for (int j = 0; j < img[0].size(); j++) {
        int size = 0;
        int sum = 0;
        for (int ti = -1; ti <= 1; ti++) {
          for (int tj = -1; tj <= 1; tj++) {
            if (i + ti < 0 || i + ti >= img.size()
              || j + tj < 0 || j + tj >= img[0].size()) {
              continue;
            }
            sum += img[i + ti][j + tj];
            size++;
          }
        }
        newImg[i][j] = sum / size;
      }
    }
    return newImg;
  }
};

File: leetcode/1easy/박은비/661.swift
import Foundation

class Leetcode.이재훈.이재훈.Solution {

    func imageSmoother(_ img: [[Int]]) -> [[Int]] {
        var answer: [[Int]] = img

        let width = img.first!.count
        let height = img.count
        let dx = [0, 1, 1, 0, -1, -1, -1, 0, 1]
        let dy = [0, 0, 1, 1, 1, 0, -1, -1, -1]

        func smoother(x: Int, y: Int) -> Int {
            var sum = 0
            var count = 0
            
            (0..<9).forEach { i in

                let (newX, newY) = (x+dx[i], y+dy[i])

                guard 0..<width ~= newX && 0..<height ~= newY else {
                    return
                }

                sum += img[newY][newX]
                count += 1

            }

            return sum/count
        }


        for i in 0..<height {
            for j in 0..<width {
                answer[i][j] = smoother(x: j, y: i)
            }
        }

        return answer
    }
}
